src/cryptonote_core/miner.h

```diff
@@ -74,6 +86,14 @@ namespace cryptonote
     void pause();
     void resume();
     void do_print_hashrate(bool do_hr);
+    bool get_is_background_mining_enabled() const;
+    bool set_is_background_mining_enabled(const bool& is_background_mining_enabled);
```

Comment:
Built-in types should not be taken by const-reference. I would recommend updating all of these functions accordingly.